src/validateChangelog.js

```diff
@@ -0,0 +1,74 @@
+const { parseChangelog } = require('./parseChangelog');
+
+/**
+ * @typedef {import('./constants.js').Version} Version
+ */
+
+/**
+ * Represents a formatting error in a changelog.
+ */
+class ChangelogFormattingError extends Error {
+  /**
+   * @param {Object} options
+   * @param {string} options.validChangelog - The string contents of the well-
+   *   formatted changelog.
+   * @param {string} options.invalidChangelog - The string contents of the
+   *   malformed changelog.
+   */
+  constructor({ validChangelog, invalidChangelog }) {
+    super('Changelog is not well-formatted');
+    this.data = {
+      validChangelog,
+      invalidChangelog,
+    };
+  }
+}
+
+/**
+ * Validates that a changelog is well-formatted.
+ * @param {Object} options
+ * @param {string} options.changelogContent - The current changelog
+ * @param {Version} options.currentVersion - The current version
+ * @param {string} options.repoUrl - The GitHub repository URL for the current
+ *   project.
+ * @param {boolean} options.isReleaseCandidate - Denotes whether the current
+ *   project is in the midst of release preparation or not. If this is set, this
+ *   command will also ensure the current version is represented in the
+ *   changelog with a release header, and that there are no unreleased changes
+ *   present.
+ */
+function validateChangelog({
+  changelogContent,
+  currentVersion,
+  repoUrl,
+  isReleaseCandidate,
+}) {
+  const changelog = parseChangelog({ changelogContent, repoUrl });
+
+  // Ensure release header exists, if necessary
+  if (
+    isReleaseCandidate &&
+    !changelog
+      .getReleases()
+      .find((release) => release.version === currentVersion)
+  ) {
+    throw new Error(
```

Comment:
Not sure on whether I should bother with a custom Error class for these other two errors as well :thinking: 

I only did that for the formatting error because I wanted to return extra information.

Reply:
I suppose one alternative would be a Result-style return value, but we don't have any conventional types for that. Returning `true` for "valid" and an error object for "invalid" seemed bad because the invalid case is still truthy, making it easy to misuse that result. But returning `false` or `null` for the valid case also seemed semantically confusing.

Reply:
I'll update it for now to use a common Error type, so that the "invalid changelog" cases can be distinguished from totally unexpected errors.

Reply:
Done in 8b592edc2498f7f2a2f023f2ea0d5b5a7f750db8